### PR TITLE
FEAT(FTDA): Setup RTK Query and hookup '/todos' endpoint with the FE to render all todo items

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**DESCRIPTION OF CHANGES**
- Set up Redux store and added RTK Query API slice for todos
- Configured base API endpoint to connect to the backend
- Implemented `useGetTodosQuery` hook to fetch todos from the backend API
- Fixed issue with `dueDate` by converting it to a readable date format
- Updated Task type property from summary to description to match the BE response 
- Successfully rendered and displayed the list of todos fetched from the database

https://github.com/user-attachments/assets/5dd4dc76-34c2-4521-b281-2393cf31ddac

### CODE REVIEW

File: services/todoApi.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+import { Task } from "@/modules/HomePage/types/Task";
+
+export const todoApi = createApi({
+  reducerPath: "todoApi",
+  baseQuery: fetchBaseQuery({ baseUrl: "http://localhost:3001" }),
```
Comment (ishmam-iqbal-sazim) [inline services/todoApi.ts]: Best practice is to put the url in your `.env` file and import that var from there.

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 2]
```diff
@@ -85,14 +86,15 @@ const TaskList: React.FC<TaskListProps> = ({
...
+      description: editFormState.description,
       dueDate: new Date(editFormState.dueDate),
       priority: editFormState.priority,
     };
     saveTask(updatedTask);
     setEditFormState({ title: "", description: "", dueDate: "", priority: 1 });
   };
-
+  const { data: todos } = useGetTodosQuery();
+  console.log("Todos", todos);
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx]: Let's remove unnecessary console logs

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 3]
```diff
@@ -85,14 +86,15 @@ const TaskList: React.FC<TaskListProps> = ({
...
-      summary: editFormState.description,
+      description: editFormState.description,
       dueDate: new Date(editFormState.dueDate),
       priority: editFormState.priority,
     };
     saveTask(updatedTask);
     setEditFormState({ title: "", description: "", dueDate: "", priority: 1 });
   };
-
+  const { data: todos } = useGetTodosQuery();
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx]: I think we should also handle isLoading from the query and display a loading indicator if its true.

File: modules/HomePage/containers/MainContainer.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { Container, Button, Group } from "@mantine/core";
 
 import TaskForm from "../components/TaskForm/TaskForm";
 import TaskList from "../components/TaskList/TaskList";
-import useTaskOperations from "@/shared/utils/useTaskOperations";
+import useTaskOperations from "@/shared/utils/usetaskOperations";
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/containers/MainContainer.tsx]: @ imports should be above relative imports

File: modules/TaskDetailsPage/TaskDetails.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ import {
 
 import classes from "./TaskDetails.module.css";
 import { Task } from "../HomePage/types/Task";
-import useTaskOperations from "@/shared/utils/useTaskOperations";
+import useTaskOperations from "@/shared/utils/usetaskOperations";
```
Comment (ishmam-iqbal-sazim) [inline modules/TaskDetailsPage/TaskDetails.tsx:13]: Same import comment as above

**Comment by yamin-faruque-sazim** [comment]
Addressed PR comments

File: .gitignore [hunk 6]
```diff
@@ -61,3 +61,6 @@ jspm_packages
 # Optional REPL history
 .node_repl_history
 .next
+
+# Env variables
+.env.local
```
Comment (ishmam-iqbal-sazim) [inline .gitignore:66]: Let's add new line here

File: modules/TaskDetailsPage/TaskDetails.tsx [hunk 7]
```diff
@@ -10,11 +9,12 @@ import {
   TextInput,
   Select,
 } from "@mantine/core";
+import useTaskOperations from "@/shared/utils/usetaskOperations";
```
Comment (ishmam-iqbal-sazim) [inline modules/TaskDetailsPage/TaskDetails.tsx:13]: Should be new line above

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 8]
```diff
@@ -16,6 +17,7 @@ import { useDisclosure } from "@mantine/hooks";
 import { Task } from "../../types/Task";
 import classes from "./TaskList.module.css";
 import { PRIORITY_OPTIONS } from "@/shared/constants/taskContstants";
+import { useGetTodosQuery } from "@/services/todoApi";
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx]: @ imports should be above relative imports

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 9]
```diff
@@ -85,14 +87,17 @@ const TaskList: React.FC<TaskListProps> = ({
...
       priority: editFormState.priority,
     };
     saveTask(updatedTask);
     setEditFormState({ title: "", description: "", dueDate: "", priority: 1 });
   };
-
+  const { data: todos, isLoading } = useGetTodosQuery();
+  if (isLoading) {
+    return <Loader size="xl" />;
+  }
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx:103]: Let's add new lines above and below

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 10]
```diff
@@ -104,7 +109,7 @@ const TaskList: React.FC<TaskListProps> = ({
         </Group>
       </Modal>
 
-      {sortedTasks.map((task) => (
+      {todos?.map((task) => (
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx:116]: What if `todos.length === 0`? Wouldn't an error be thrown inside the map loop?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: modules/HomePage/components/TaskList/TaskList.tsx (hunk 10 above)
Comment (yamin-faruque-sazim) [reply modules/HomePage/components/TaskList/TaskList.tsx:116]: Are you saying, if there are no todo item in the db, I might get an error here with this implementation? How I am understanding is, if todos.length ===0, that means there are no items in that array, ie no todos in the db. So no todos would be rendered.

**Pull request merged**